hist/histpainter/src/TGraphPainter.cxx

```diff
@@ -4365,6 +4365,22 @@ void TGraphPainter::PaintScatter(TScatter *theScatter, Option_t* chopt)
          maxs = TMath::Max(maxs,theSize[i]);
       }
    }
+   if (minx == maxx) {
+      minx = 0.9*theX[0];
+      maxx = 1.1*theX[0];
```

Comment:
The code seems to be duplicated below for `miny`, `maxy`, `theY`, etc.   Perhaps, we could instead define a lambda expression as in
```c++
auto fnFixRange = [](double &min, double &max, double value) {
   // ...
};
```
and call it accordingly for all the cases below?

Also, could you add a comment on why these values need to be adjusted?

Reply:
That's super simple protection when min==max ... Which in general is not a good idea, and, in that case, produces wrong results... I guess the code is self-explanatory. A lambda will be a bit confusing for no real added value seems to me.